src/get-started/java-binding.md

```diff
@@ -0,0 +1,382 @@
+the message) - `isValid` method; and the execution rule for the
+transaction - `execute` method.
+
+Ed25519 is a standard cryptographic system for digital signature of Exonum
+messages. We use `com.exonum.binding.crypto.CryptoFunctions#ed25519`
+method for obtaining the implementation of the function.
+
+The implementation of `Transaction#isValid` transaction authentication method
+must be a pure function, i.e. for the given transaction to return the same
+result on all nodes of the system. For this reason, access to the Exonum
+storage by external files or network resources in method implementations is
+not allowed.
+
+`Transaction#execute` method describes the operations that are applied to the
+current storage state when the transaction is executed. Exonum transfers `Fork`
+as an argument - a view that allows performing modifying operations. A service
+schema object can be used to access data collections of this service.
+
+!!! note
+    `Transaction#info` will be deleted soon.
+
+For more details about transactions in Exonum - their properties and processing
+rules - see our [documentation][transactions]
+
+### Description of the External Service API
+
+The external service API serves for interaction of the service and the external
+systems. A set of operations is defined by each service and can include sending
+transactions to the network, read requests for the blockchain data with the
+provision of the cryptographic proof, etc. Exonum provides an embedded web
+framework for describing the REST-interface of the service.
+
+`com.exonum.binding.service.Service#createPublicApiHandlers` method is used to
+determine the handlers for HTTP requests. These handlers are available at the
+common path corresponding to the service name. Thus, the `/balance/:walletId`
+handler for balance requests in the "cryptocurrency" service will be available
+at `/cryptocurrency/balance/:walletId`.
+
+See [documentation][vertx.io] on the possibilities of `Vert.x` used as a web
+framework.
+
+### Dependencies Management
+
+To describe the dependencies of the service components (both system-specific
+ones, for example, Exonum time service, and external ones), Guice is used. Each
+service should define a module describing implementations of the framework
+interfaces - `com.exonum.binding.service.Service`,
+`com.exonum.binding.service.TransactionConverter` and other components
+implementations, if any.
+
+!!! note "Minimalistic Example of Service Module"
+    ```java
+    public class ServiceModule extends AbstractModule {
+
+      @Override
+      protected void configure() {
+        // Define the Service implementation.
+        bind(Service.class).to(CryptocurrencyService.class).in(Singleton.class);
+
+        // Define the TransactionConverter implementation.
+        bind(TransactionConverter.class).to(CryptocurrencyTransactionConverter.class);
+      }
+    }
+    ```
+
+The full name of the module class must be submitted when the service launches.
```

Comment:
The fully-qualified name (or FQN) of the module class must be specified/passed during configuration of an Exonum App that will run the service.

?